frontend/src/metabase/dashboard/components/DashboardHeader/DashboardHeaderButtonRow.tsx

```diff
@@ -0,0 +1,315 @@
+  AddTemporalUnitButton,
+  CopyAnalyticsDashboardButton,
+  DashboardActionMenu,
+  DashboardInfoButton,
+  DashboardSubscriptionButton,
+  EditDashboardButton,
+  FullscreenAnalyticsDashboard,
+  getExtraButtons,
+  shouldRenderSubscriptionButton,
+  FullscreenToggle,
+  NightModeToggleButton,
+} from "./buttons";
+
+export const BUTTON_CONFIG = {
+  ADD_QUESTION: "ADD_QUESTION",
+  ADD_HEADING_OR_TEXT: "ADD_HEADING_OR_TEXT",
+  ADD_LINK_CARD: "ADD_LINK_CARD",
+  ADD_SECTION: "ADD_SECTION",
+  ADD_TEMPORAL_UNIT: "ADD_TEMPORAL_UNIT",
+  ADD_FILTER_PARAMETER: "ADD_FILTER_PARAMETER",
+  ADD_ACTION_ELEMENT: "ADD_ACTION_ELEMENT",
+  EXTRA_EDIT_BUTTONS_MENU: "EXTRA_EDIT_BUTTONS_MENU",
+  COPY_ANALYTICS_DASHBOARD: "COPY_ANALYTICS_DASHBOARD",
+  EDIT_DASHBOARD: "EDIT_DASHBOARD",
+  DASHBOARD_SUBSCRIPTION: "DASHBOARD_SUBSCRIPTION",
+  DASHBOARD_EMBED_ACTION: "DASHBOARD_EMBED_ACTION",
+  REFRESH_WIDGET: "REFRESH_WIDGET",
+  NIGHT_MODE_TOGGLE: "NIGHT_MODE_TOGGLE",
+  FULLSCREEN_TOGGLE: "FULLSCREEN_TOGGLE",
+  DASHBOARD_HEADER_ACTION_DIVIDER: "DASHBOARD_HEADER_ACTION_DIVIDER",
+  DASHBOARD_BOOKMARK: "DASHBOARD_BOOKMARK",
+  DASHBOARD_INFO: "DASHBOARD_INFO",
+  DASHBOARD_ACTION_MENU: "DASHBOARD_ACTION_MENU",
+  FULLSCREEN_ANALYTICS_DASHBOARD: "FULLSCREEN_ANALYTICS_DASHBOARD",
+} as const;
+
+type ButtonKey = keyof typeof BUTTON_CONFIG;
+
+type DashboardHeaderButtonRowProps = WithRouterProps & {
+  collection?: Collection;
+  isEmpty?: boolean;
+  isPublic?: boolean;
+} & DashboardRefreshPeriodControls &
+  DashboardFullscreenControls &
+  Pick<
+    EmbedThemeControls,
+    "hasNightModeToggle" | "isNightMode" | "onNightModeChange"
+  >;
+
+type HeaderButtonProps = {
+  isEditing: boolean;
+  canEdit: boolean;
+  hasModelActionsEnabled: boolean;
+  isAnalyticsDashboard: boolean;
+  dashboard: Dashboard;
+  canManageSubscriptions: boolean;
+  formInput: any;
+  isAdmin: boolean;
+} & DashboardHeaderButtonRowProps;
+
+type ButtonConfig = {
+  component: (props: HeaderButtonProps) => ReactNode;
+  enabled: (props: HeaderButtonProps) => boolean;
+};
+
+const buttonConfigs: Record<ButtonKey, ButtonConfig> = {
+  [BUTTON_CONFIG.ADD_QUESTION]: {
+    component: () => <AddQuestionButton />,
+    enabled: ({ isEditing }) => isEditing,
+  },
+  [BUTTON_CONFIG.ADD_HEADING_OR_TEXT]: {
+    component: () => <AddHeadingOrTextButton />,
+    enabled: ({ isEditing }) => isEditing,
+  },
+  [BUTTON_CONFIG.ADD_LINK_CARD]: {
+    component: () => <AddLinkCardButton />,
+    enabled: ({ isEditing }) => isEditing,
+  },
+  [BUTTON_CONFIG.ADD_SECTION]: {
+    component: () => <AddSectionButton />,
+    enabled: ({ isEditing }) => isEditing,
+  },
+  [BUTTON_CONFIG.ADD_TEMPORAL_UNIT]: {
+    component: () => <AddTemporalUnitButton />,
+    enabled: ({ isEditing }) => isEditing,
+  },
+  [BUTTON_CONFIG.ADD_FILTER_PARAMETER]: {
+    component: () => <AddFilterParameterButton />,
+    enabled: ({ isEditing }) => isEditing,
+  },
+  [BUTTON_CONFIG.ADD_ACTION_ELEMENT]: {
```

Comment:
We missed a divider here
#### stats
![image](https://github.com/user-attachments/assets/920d15e5-f05b-4d81-8da0-0a47b5924e49)

#### This PR
![image](https://github.com/user-attachments/assets/badcd6c3-a203-42a8-a435-2ca1f1ab70bb)